### PR TITLE
Bugfix/36 gateway allows access to deleted api keys

### PR DESCRIPTION
**Pull request by BrunoSilvaATB**
- Now when a call to the gateway is made with a deleted API key, it returns a 403 Forbidden.

### CODE REVIEW

File: spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java [hunk 1]
```diff
@@ -140,6 +141,9 @@ private boolean isApiKeyValid(String apiKeyValue) throws AuthenticationException
                 if (Date.valueOf(LocalDate.now()).after(apiKey.getValidUntil())) {
                     throw new CredentialsExpiredException("Api key is expired");
                 }
+                if (apiKey.getDeletedWhen() != null) {
+                    throw new AccessDeniedException("access denied");
```
Comment (gr-hovest-atb) [inline spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java:145]: Suggestion: return false here, then a BadCredentialsException will be thrown.

**Review by gr-hovest-atb — CHANGES_REQUESTED** [review]

File: spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java (hunk 1 above)
Comment (BrunoSilvaATB) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java]: I changed and tested.
It is working well.

Comment (BrunoSilvaATB) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java]: As soon as you accepted it, I will proceed to make the PR to the Main repository

Comment (coatrack-test-gghvst) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java]: Looks good!

Comment (gr-hovest-atb) [reply spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java]: OK!

**Review by gr-hovest-atb — APPROVED** [review]

**Pull request closed without merging**